src/backend/distributed/shared_library_init.c

```diff
@@ -670,6 +670,21 @@ RegisterCitusConfigVariables(void)
 		GUC_NO_SHOW_ALL,
 		NULL, NULL, NULL);
 
+	DefineCustomBoolVariable(
+		"citus.allow_unsafe_locks_from_workers",
```

Comment:
is there any reason why we would want to allow this?

It seems better to throw an error telling users to add the coordinator to the metadata, since we don't really want to support clusters without it.

Reply:
Mostly not to break existing users who only queries from workers and keep regression test changes manageable. Otherwise we end with a quite large PR -- even comparable to this one: https://github.com/citusdata/citus/pull/5756. I think once we implement 5756, we could drop this GUC as well?

Reply:
Aaah, for truncate, got it.

Reply:
Oh, yes, the primary goal is TRUNCATE. Well, thanks for commenting on this, now the reason is documented on Github as well:)